### PR TITLE
Add deploy plugin blueprint files

### PR DESCRIPTION
**Pull request by achambers**

### CODE REVIEW

File: files/.travis.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+---
+language: node_js
+node_js:
+  - "6"
```
Comment (lukemelia) [inline files/.travis.yml]: Add `- "4"`

Comment (achambers) [reply files/.travis.yml]: 👍

File: files/tests/index-test.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+/*eslint-env node*/
+'use strict';
+
+var chai  = require('chai');
+var chaiAsPromised = require("chai-as-promised");
+chai.use(chaiAsPromised);
+
+var assert = chai.assert;
```
Comment (lukemelia) [inline files/tests/index-test.js]: We should make a helper for assert and require it, like we do in some repos.

File: files/tests/index-test.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+describe('<%= shortName %> plugin', function() {
+  it('does something', function() {
+    assert.isOk(false);
+  });
```
Comment (lukemelia) [inline files/tests/index-test.js]: showing example of testing a hook is something we should consider here

File: files/tests/runner.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+function addFiles(mocha, files) {
+  glob.sync(root + files).forEach(mocha.addFile.bind(mocha));
+}
+
+addFiles(mocha, '/**/*-test.js');
+
+if (arg === 'all') {
+  addFiles(mocha, '/**/*-test-slow.js');
+}
```
Comment (lukemelia) [inline files/tests/runner.js]: I don't think we actually use this part at all.

File: files/tests/index-test.js (hunk 2 above)
Comment (achambers) [reply files/tests/index-test.js]: 👍

File: files/tests/runner.js (hunk 4 above)
Comment (achambers) [reply files/tests/runner.js]: That's accurate. I'll remove

File: files/tests/index-test.js (hunk 3 above)
Comment (achambers) [reply files/tests/index-test.js]: hmm...Yeh...I considered that briefly. I'll explore it deeper.

File: files/index.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+      requiredConfig: ['isInNeedOfSleep'], // Example required config. Remove this;
+
+      /*
+       * Implement any pipeline hooks here
+       *
+       * http://ember-cli-deploy.com/docs/v1.0.x/pipeline-hooks/
+       */
+
+      //configure: function(context) {
+      //  var configProp = this.readConfig('foo'); // this is how you acceess plugin config
```
Comment (lukemelia) [inline files/index.js]: typo `access`

File: files/index.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+      //  return {
+      //    someProp: 'someValue'
+      //  };
+      //},
+
+      //willDeploy: function(context) {
+      //  return RSVP.resolve(); // Return a promise if you'd like the pipeline to wait until the hook has performed some function
+      //},
+
+      //willBuild: function(context) {},
```
Comment (lukemelia) [inline files/index.js]: Can we use `willBuild() {}` as of node 4?

Comment (achambers) [reply files/index.js]: Will check. Would be nice to start it out with es2015 syntax if possible.

**Pull request merged**